File: groupe/src/main/java/unihelp/example/groupe/entities/Groupe.java
package unihelp.example.groupe.entities;


import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.Date;
import java.util.HashSet;
import java.util.Set;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter

public class Groupe
{
    @Id
    @GeneratedValue()
    private Long groupId;
    private String groupName;
    @OneToMany(mappedBy = "groupe", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference
    private Set<GroupMembership> members = new HashSet<>();
    private String createdBy;
    private Long createdById;     // ✅ l’ID à ajouter si pas encore fait

    @OneToMany(mappedBy = "groupe", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonBackReference
    private Set<JoinRequest> joinRequests = new HashSet<>();
    @Transient
    @JsonInclude(JsonInclude.Include.NON_NULL)
    private Integer messageCount;
    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "chat_id", referencedColumnName = "chatId")
    private Chat chat;
    @Column(length = 1000)
    private String description;
    @Column(name = "created_at")
    private LocalDateTime createdAt;
    @Column(name = "group_image", columnDefinition = "LONGTEXT")
    private String groupImage;


}

File: frontend/src/app/models/groupe.ts
import { Chat } from './chat';
import { User } from './user.model';
import { GroupMemberDTO } from './GroupeMemberDTO';

export interface Groupe {
  groupId: number;
  groupName: string;
  users?: User[];
  createdBy: string;
  createdById: number; // ✅ Identifiant du créateur

  chat?: Chat;
  groupImage?: string; // 👈 pour l'image base64
  messageCount?: number; // ✅
  members?: GroupMemberDTO[];
  createdAt?: string | Date | null;
  description?: string;
  // ✅ Ajout ici

}

